### PR TITLE
Monitoring start and end times

### PR DESCRIPTION
**Pull request by devarrowsmith**
These changes fulfil the requirements of [ELM-3048](https://dsdmoj.atlassian.net/jira/software/c/projects/ELM/boards/1659?selectedIssue=ELM-3048):

- Adds a dateTimeField component, that accepts date and time values to generate a datetime.
- The dateTimeField is used in the monitoring conditions page to capture the order start and end datetimes.
- Start datetime is a mandatory field. End datetime is optional. Both have appropriate validation & error messages.

Additional improvements:
- The monitoring conditions page has been refactored. It follows modular patterns used for other form sections. 
- Validation of order start and end times, and other fields in the monitoring conditions page, is performed on the frontend using Zod in the first instance. Backend validation errors are still handled correctly once frontend validation errors have been resolved.
- Tests are updated. Cypress tests make more use of helper functions.

[ELM-3048]: https://dsdmoj.atlassian.net/browse/ELM-3048?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: integration_tests/e2e/order/monitoring-conditions/monitoringConditions.cy.ts [hunk 1]
```diff
@@ -90,31 +94,61 @@ context('Monitoring conditions main section', () => {
...
+      cy.get('#endDate-year').type('2025')
+      cy.get('#endDate-hours').type('03')
+      cy.get('#endDate-minutes').type('04')
+      page.form.saveAndContinueButton.click()
+      cy.get('#orderType-error').should('contain', 'Test error - order type')
+      cy.get('#orderTypeDescription-error').should('contain', 'Test error - order type description')
+      cy.get('#conditionType-error').should('contain', 'Test error - condition type')
+      cy.get('#monitoringRequired-error').should('contain', 'Test error - monitoring required')
+      cy.get('#startDate-error').should('contain', 'Test error - start date')
+      cy.get('#endDate-error').should('contain', 'Test error - end date')
```
Comment (ben-cart3r) [inline integration_tests/e2e/order/monitoring-conditions/monitoringConditions.cy.ts:147]: The MonitoringConditionsPage has helpers for you to do all of these kinds of things. So you could do 

```js
page.form.fillInWith({
  orderType: 'immigration'
  // add the other fields here
})
```
similarly
```js
page.form.startDateField.shouldHaveValidationMessage('Test error - order type')
```

File: integration_tests/e2e/order/monitoring-conditions/monitoringConditions.cy.ts [hunk 2]
```diff
@@ -130,17 +164,22 @@ context('Monitoring conditions main section', () => {
...
+      cy.get('#endDate-month').type('4')
+      cy.get('#endDate-year').type('2025')
+      cy.get('#endDate-hours').type('03')
+      cy.get('#endDate-minutes').type('04')
       page.form.saveAndContinueButton.click()
       Page.verifyOnPage(InstallationAddressPage)
       cy.task('getStubbedRequest', `/orders/${mockOrderId}/monitoring-conditions`).then(requests => {
         expect(requests).to.have.lengthOf(1)
         expect(requests[0]).to.deep.equal({
+          action: 'continue',
```
Comment (ben-cart3r) [inline integration_tests/e2e/order/monitoring-conditions/monitoringConditions.cy.ts:178]: I don't think we need to send `action` to the api

Comment (devarrowsmith) [reply integration_tests/e2e/order/monitoring-conditions/monitoringConditions.cy.ts]: `action` is no longer sent to the api.

**Comment by devarrowsmith** [comment]
Along with the changes requested in the comments, the following changes have been made:
- Obsolete `startTime` & `endTime` properties in test files have been removed. 
- Handling of dateTime validation errors has been improved. Their rendering in the dateTimeInput template has also been improved.

**Review by ben-cart3r — APPROVED** [review]
👍

**Pull request merged**